### DISCUSSION TITLE
Different output on generate note between API/UI

### DISCUSSION THREAD

**Discussion by benoittgt** [community/community#45320] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

Hello

I am trying to do via API the same thing I do with the UI. Generate a release note with a specific target (not the HEAD on master). On the UI the changelog is properly generated but not with the API.

![Screenshot_2023-01-25_at_11_27_13](https://user-images.githubusercontent.com/8417720/214541351-0dbcc9ea-95b4-43e0-a366-7235a49d40ac.jpg)

With the API
```sh
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  /repos/xxxxxx/echoserver/releases/generate-notes \
  -f tag_name='v0.2.1' \
 -f target_commitish='dec20d6cf0dd65c621bfeec32c158d6xbbea9f55' \
 -f previous_tag_name='v0.2.0'
```
Output
```json
{
  "name": "v0.2.1",
  "body": "**Full Changelog**: https://github.com/xxxxxx/echoserver/compare/v0.2.0...v0.2.1"
}
```

Body is incomplet, I do not see the pull request merged. Do I miss something?

**Comment by jge162** [comment]
What I see is the API call you are using is generating a release note with the name and body provided in the API request. The "body" field contains a link to the full changelog between the specified tags, which is why you don't see the pull requests listed in the body.

**Comment by benoittgt** [comment]
Sorry it is not super clear. The body is the output. If I remove the target commit.

```sh
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  /repos/xxxxx/echoserver/releases/generate-notes \
  -f tag_name='v0.2.1' \
 -f previous_tag_name='v0.2.0'
```
Output
```json
{
  "name": "v0.2.1",
  "body": "## What's Changed\n* feat(xxxx-397): Add a commit to test @benoittgt in https://github.com/xxxx/echoserver/pull/27\n\n\n**Full Changelog**: https://github.com/xxxx/echoserver/compare/v0.2.0...v0.2.1"
}
```

If I specified a commit that is the HEAD minus one.
```sh
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  /repos/xxxx/echoserver/releases/generate-notes \
  -f tag_name='v0.2.1' \
 -f target_commitish='f4a279e0f88bf17983a01a32c81427f70507432d' \
 -f previous_tag_name='v0.2.0'
```
Output
```json
{
  "name": "v0.2.1",
  "body": "**Full Changelog**: https://github.com/xxxx/echoserver/compare/v0.2.0...v0.2.1"
}
```

**Reply by jge162** [reply]
> Sorry it is not super clear. The body is the output. If I remove the target commit.
> 
> ```shell
> gh api \
>   --method POST \
>   -H "Accept: application/vnd.github+json" \
>   /repos/xxxxx/echoserver/releases/generate-notes \
>   -f tag_name='v0.2.1' \
>  -f previous_tag_name='v0.2.0'
> ```
> 
> Output
> 
> ```json
> {
>   "name": "v0.2.1",
>   "body": "## What's Changed\n* feat(xxxx-397): Add a commit to test @benoittgt in https://github.com/xxxx/echoserver/pull/27\n\n\n**Full Changelog**: https://github.com/xxxx/echoserver/compare/v0.2.0...v0.2.1"
> }
> ```
> 
> If I specified a commit that is the HEAD minus one.
> 
> ```shell
> gh api \
>   --method POST \
>   -H "Accept: application/vnd.github+json" \
>   /repos/xxxx/echoserver/releases/generate-notes \
>   -f tag_name='v0.2.1' \
>  -f target_commitish='f4a279e0f88bf17983a01a32c81427f70507432d' \
>  -f previous_tag_name='v0.2.0'
> ```
> 
> Output
> 
> ```json
> {
>   "name": "v0.2.1",
>   "body": "**Full Changelog**: https://github.com/xxxx/echoserver/compare/v0.2.0...v0.2.1"
> }
> ```


Bear with me here please. I’m trying my best here. So, It seems that the target_commitish parameter is specifying a specific commit, rather than the commit that is the HEAD minus one. I see in the case above. 

The target_commitish parameter is used to specify a specific commit or branch to use as the base for the comparison, rather than the default branch!

But @benoittgt If you want to compare the commit that is the HEAD minus one, maybe you can try specifying the SHA of the commit that is the HEAD minus one instead of the commit SHA in the target_commitish parameter??

 try using a HEAD~1 as the target_commitish, which references the commit before the current HEAD, as an alternative. I think I’m brain dead after this one. Hope it works.

**Comment by benoittgt** [comment, marked as answer]
Hello and thanks for the answer I find the issue.

Release note contains mentions of merged pull request. My issue is the only pull request merged I had was on the last commit so not the one I was choosing.

So there is no issue.
